code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -0,0 +1,768 @@
+			the_component = new/obj/item/clockwork/component/belligerent_eye(get_turf(src))
+			clockwork_component_cache["belligerent_eye"]--
+		if("Vanguard Cogwheel")
+			the_component = new/obj/item/clockwork/component/vanguard_cogwheel(get_turf(src))
+			clockwork_component_cache["vanguard_cogwheel"]--
+		if("Guvax Capacitor")
+			the_component = new/obj/item/clockwork/component/guvax_capacitor(get_turf(src))
+			clockwork_component_cache["guvax_capacitor"]--
+		if("Replicant Alloy")
+			the_component = new/obj/item/clockwork/component/replicant_alloy(get_turf(src))
+			clockwork_component_cache["replicant_alloy"]--
+		if("Hierophant Ansible")
+			the_component = new/obj/item/clockwork/component/hierophant_ansible(get_turf(src))
+			clockwork_component_cache["hierophant_ansible"]--
+	if(the_component)
+		user.visible_message("<span class='notice'>[user] withdraws [the_component] from [src].</span>", "<span class='notice'>You withdraw [the_component] from [src].</span>")
+		user.put_in_hands(the_component)
+	return 1
+
+/obj/structure/clockwork/cache/examine(mob/user)
+	..()
+	if(is_servant_of_ratvar(user) || isobserver(user))
+		user << "<b>Stored components:</b>"
+		user << "<i>Belligerent Eyes:</i> [clockwork_component_cache["belligerent_eye"]]"
+		user << "<i>Vanguard Cogwheels:</i> [clockwork_component_cache["vanguard_cogwheel"]]"
+		user << "<i>Guvax Capacitors:</i> [clockwork_component_cache["guvax_capacitor"]]"
+		user << "<i>Replicant Alloys:</i> [clockwork_component_cache["replicant_alloy"]]"
+		user << "<i>Hierophant Ansibles:</i> [clockwork_component_cache["hierophant_ansible"]]"
+
+/obj/structure/clockwork/ocular_warden //Ocular warden: Low-damage, low-range turret. Deals constant damage to whoever it makes eye contact with.
+	name = "ocular warden"
+	desc = "A large brass eye with tendrils trailing below it and a wide red iris."
+	clockwork_desc = "A stalwart turret that will deal sustained damage to any non-faithful it sees."
+	icon_state = "ocular_warden"
+	health = 25
+	max_health = 25
+	construction_value = 15
+	break_message = "<span class='warning'>The warden's eye gives a glare of utter hate before falling dark!</span>"
+	debris = list(/obj/item/clockwork/component/replicant_alloy/blind_eye)
+	var/damage_per_tick = 3
+	var/sight_range = 3
+	var/mob/living/target
+
+/obj/structure/clockwork/ocular_warden/New()
+	..()
+	SSfastprocess.processing += src
+
+/obj/structure/clockwork/ocular_warden/Destroy()
+	SSfastprocess.processing -= src
+	return ..()
+
+/obj/structure/clockwork/ocular_warden/examine(mob/user)
+	..()
+	user << "[target ? "It's fixated on [target]" : "Its gaze is wandering aimlessly"]."
+
+/obj/structure/clockwork/ocular_warden/process()
+	if(ratvar_awakens && (damage_per_tick == initial(damage_per_tick) || sight_range == initial(sight_range))) //Massive buff if Ratvar has returned
+		damage_per_tick = 10
+		sight_range = 5
+	if(target)
+		if(target.stat || get_dist(get_turf(src), get_turf(target)) > sight_range)
```

Comment:
Should also check if the target is a servant, ie; it got converted while it was attacking